File: 2869-minimum-operations-to-collect-elements/2869-minimum-operations-to-collect-elements.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        count = 0    
        res = []
        rng = range(1, k + 1)
        
        while not all(x in res for x in rng):
            res.append(nums.pop())
            count += 1
        
        return count
        

File: 2869-minimum-operations-to-collect-elements/2869-minimum-operations-to-collect-elements.rs
use std::collections::HashSet;
impl Solution {
    pub fn min_operations(mut nums: Vec<i32>, k: i32) -> i32 {
        let mut count = 0;
        let mut rng: HashSet<i32> = (1..=k).collect();

        while !rng.is_empty() {
            if let Some(val) = nums.pop() {
                rng.remove(&val);
                count += 1;
            }
        }

        count
    }
}
